lib/mediakit/utils/constant_class_definer.rb

```diff
@@ -0,0 +1,41 @@
+require 'active_support/core_ext/module/introspection'
+
+module Mediakit
+  module Utils
+    module ConstantClassDefiner
+      def self.included(base)
+        base.extend(ClassMethods)
+      end
+
+      module ClassMethods
+        def using_attributes
+          raise(NotImplementedError)
+        end
+
+        def define_subclass(klass, attributes)
+          raise(
+            ArgumentError,
+            <<EOS
+you can give attribute keys which only defined by `using_attributes` and that is satisfied all keys.'
+EOS
+          ) unless Set.new(using_attributes) == Set.new(attributes.keys)
+
+          klass_name = klass.to_s
+          return if parent.const_defined?(klass_name)
+
+          parent.const_set(
```

Comment:
Maybe, I should use the singleton object based implementation instead of the Class object based implementation.